coder/packages
# Prevent DB migration errors when deploying via Render

- In https://github.com/coder/coder/issues/6841, users have reported issues when deploying Coder via Render
- When deploying Coder via Render currently, Render will start the Coder application then quickly restart the Coder application to provide a new environment variable. This interrupts Coder while it is performing a DB migration, leaving the DB in a dirty state requiring manual intervention.
- This PR removes reliance upon `RENDER_HOST`, an environment variable injected via a Render Blueprint, in favor of `RENDER_EXTERNAL_URL`, an environment variable that is set in all Render web services. Removal of `RENDER_HOST` means that Render does not need to quickly restart the Coder application.

Testing:
- On a Render account with a credit card attached, create a new blueprint and enter into the Public Git Repository my fork: https://github.com/scottnuma/packages
- Select the `scott/dag` branch and create resources
- Monitor the service logs and observe the migration error logs no longer appear

Alternative solutions that were not chosen:
- Change the migrations to better tolerate failures. In particular user reproductions and my reproductions repeatedly see error messages from this migration: `CREATE INDEX CONCURRENTLY workspace_resources_job_id_idx ON workspace_resources USING btree (job_id);`. Creating indexes with `CONCURRENTLY` has some interesting footguns and its failures leave manual clean up of invalid indexes - [blogpost](https://about.sourcegraph.com/blog/introducing-migrator-service)
- Trigger the migration manually rather than automatically. This could be decoupling automatic migration from Coder start up or requiring users to manually enter the DB connection string when launching on Render.
